sphinx_gallery/docs_resolv.py

```diff
@@ -216,66 +163,46 @@ def __init__(self, doc_url, gallery_dir, searchindex='searchindex.js',
         else:
             self._is_windows = False
 
+        # Download and find documentation options.
+        index = get_data(index_url, gallery_dir)
+        self._docopts = parse_sphinx_docopts(index)
+
         # download and initialize the search index
         sindex = get_data(searchindex_url, gallery_dir)
-        filenames, objects = parse_sphinx_searchindex(sindex)
-
-        self._searchindex = dict(filenames=filenames, objects=objects)
+        self._searchindex = js_index.loads(sindex)
 
     def _get_link(self, cobj):
         """Get a valid link, False if not found"""
 
-        fname_idx = None
-        full_name = cobj['module_short'] + '.' + cobj['name']
-        if full_name in self._searchindex['objects']:
-            value = self._searchindex['objects'][full_name]
-            if isinstance(value, dict):
-                value = value[next(iter(value.keys()))]
```

Comment:
I just tested this PR. but in this case it does not even recognize `np.random.random` I'll see with #273